lib/SILOptimizer/IPO/DeadFunctionElimination.cpp

```diff
@@ -421,7 +429,15 @@ class DeadFunctionAndGlobalElimination {
     findAnchorsInTables();
 
     for (SILFunction &F : *Module) {
-      if (isAnchorFunction(&F)) {
+      // In embedded Swift, generic functions, even public ones cannot be used
+      // externally and are not anchors.
+      bool embedded =
+          Module->getASTContext().LangOpts.hasFeature(Feature::Embedded);
+      bool generic = loweredFunctionHasGenericArguments(&F);
+      bool ignoreAnchor =
+          embedded && generic && removeUnspecializedFunctionsInEmbeddedSwift;
```

Comment:
IMO, it would be better to move this logic into `isAnchorFunction`

Reply:
👍